src/false_positives.h

```diff
@@ -0,0 +1,93 @@
+    {0, 0, 1, 2, 4, 6, 9, 12, 15, 18, 21, 25, 28, 32, 36, 40, 44, 48, 52, 56, 61, 65, 69, 74, 79, 83, 88, 93, 98, 102, 107, 112, 117, 122, 127, 133, 138, 143, 148, 153, 159, 164, 170, 175, 180, 186, 191, 197, 203, 208, 214, 220, 225, 231, 237, 243, 248, 254, 260, 266, 272, 278, 284, 290, 296, 302, 308, 314, 320, 326, 332, 339, 345, 351, 357, 363, 370, 376, 382, 389, 395, 401, 408, 414, 420, 427, 433, 440, 446, 453, 459, 466, 472, 479, 485, 492, 499, 505, 512, 518, 525, 532, 539, 545, 552, 559, 565, 572, 579, 586, 593, 599, 606, 613, 620, 627, 634, 640, 647, 654, 661, 668, 675, 682, 689, 696, 703, 710, 717, 724, 731},
+    {0, 0, 1, 2, 4, 6, 9, 12, 15, 18, 21, 25, 28, 32, 36, 40, 44, 48, 52, 56, 61, 65, 69, 74, 79, 83, 88, 93, 97, 102, 107, 112, 117, 122, 127, 132, 138, 143, 148, 153, 159, 164, 169, 175, 180, 186, 191, 197, 203, 208, 214, 219, 225, 231, 237, 242, 248, 254, 260, 266, 272, 278, 284, 290, 296, 302, 308, 314, 320, 326, 332, 338, 344, 351, 357, 363, 369, 376, 382, 388, 395, 401, 407, 414, 420, 427, 433, 439, 446, 452, 459, 465, 472, 478, 485, 492, 498, 505, 511, 518, 525, 531, 538, 545, 551, 558, 565, 572, 578, 585, 592, 599, 606, 612, 619, 626, 633, 640, 647, 654, 661, 668, 674, 681, 688, 695, 702, 709, 716, 723, 730},
+    {0, 0, 1, 2, 4, 6, 9, 12, 15, 18, 21, 25, 28, 32, 36, 40, 44, 48, 52, 56, 61, 65, 69, 74, 79, 83, 88, 93, 97, 102, 107, 112, 117, 122, 127, 132, 138, 143, 148, 153, 159, 164, 169, 175, 180, 186, 191, 197, 202, 208, 214, 219, 225, 231, 237, 242, 248, 254, 260, 266, 272, 278, 284, 290, 296, 302, 308, 314, 320, 326, 332, 338, 344, 351, 357, 363, 369, 376, 382, 388, 394, 401, 407, 414, 420, 426, 433, 439, 446, 452, 459, 465, 472, 478, 485, 491, 498, 505, 511, 518, 524, 531, 538, 545, 551, 558, 565, 571, 578, 585, 592, 599, 605, 612, 619, 626, 633, 640, 646, 653, 660, 667, 674, 681, 688, 695, 702, 709, 716, 723, 730},
+};
+
+size_t compute_capacity(uint32_t bits, size_t max_elements, uint32_t fpbits) {
+    if (bits == 0) return 0;
+    if (bits == 1) return max_elements + fpbits;
+    uint16_t base_fpbits = BASE_FPBITS[std::min(MAX_BITS, bits) - 2][std::min(max_elements, MAX_ELEM)];
+    if (base_fpbits >= fpbits) return max_elements;
+    return max_elements + (fpbits - base_fpbits + bits - 1) / bits;
+}
+
+size_t compute_max_elements(uint32_t bits, size_t capacity, uint32_t fpbits) {
+    size_t max_elements = capacity;
+    while (true) {
+        size_t capacity_for_max_elements = compute_capacity(bits, max_elements, fpbits);
+        CHECK_SAFE(capacity_for_max_elements >= capacity);
+        if (capacity_for_max_elements <= capacity) return max_elements;
+        size_t adjust = capacity_for_max_elements - capacity;
+        if (max_elements < adjust) return 0;
```

Comment:
nit: this can be `<=` and exit a little faster?

Reply:
I think so, yes, but that may be less clear (see added comments).